exporters/prometheus/CMakeLists.txt

```diff
@@ -16,40 +16,73 @@ include_directories(include)
 if(NOT TARGET prometheus-cpp::core)
   find_package(prometheus-cpp CONFIG REQUIRED)
 endif()
+if(WITH_METRICS_PREVIEW)
```

Comment:
We also need to modify the root [CMakeLists.txt](https://github.com/open-telemetry/opentelemetry-cpp/blob/bf8f433cae41ad682cf9e390650158a2751373c1/exporters/CMakeLists.txt#L22) to build prometheus even if WITH_METRICS_PREVIEW is not enabled?

Also need to modify test C[MakeLists.txt](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/exporters/prometheus/test/CMakeLists.txt) to build the tests only if WITH_METRICS_PREVIEW is enabled.

Also can we raise a github issue to add unit-tests so that we don't miss adding them?